app/events/notification_event.rb

```diff
@@ -1,9 +1,11 @@
+# frozen_string_literal: true
+
 class NotificationEvent < BaseEventStoreEvent
 
   # Create Notification Event
 
   def self.create(system_name, event)
-    provider_id = event.provider.try!(:id)
+    provider_id = (event.provider || Account.new({id: event.metadata[:provider_id]}, without_protection: true)).id
```

Comment:
why building an `Account` and then taking just `id` from it?

and why just `event.provider`? i mean `provider_id` would contain either `provider` or `provider ID`?

Reply:
> why building an Account and then taking just id from it?

No need 😄 


> and why just event.provider? i mean provider_id would contain either provider or provider ID?

Because depending on the kind of event it contains either the `provider` (but not provider_id) or the `provider_id` but not provider.

Reply:
then `event.provider.try(:id) || event.provider_id` ?

Reply:
That wouldn't work 😄 But I get the idea then 😉 I should do this instead:
`event.provider.try(:id) || event.metadata[:provider_id]`